aarch64_extension.cpp

```diff
@@ -586,6 +734,135 @@ class AArch64ArchitectureExtension : public ArchitectureHook {
+      return false;
+    }
+
+    if (detail->operands[0].type != ARM64_OP_REG ||
+        -1 == detail->operands[0].vector_index ||
+        ARM64_VAS_INVALID == detail->operands[0].vas ||
+        detail->operands[1].type != ARM64_OP_REG) {
+      return false;
+    }
+
+    uint32_t Rd = this->m_base->GetRegisterByName(
+        cs_reg_name(disassembler.Get(), detail->operands[0].reg));
+    uint32_t Rn = this->m_base->GetRegisterByName(
+        cs_reg_name(disassembler.Get(), detail->operands[1].reg));
+
+    size_t Rn_size = this->m_base->GetRegisterInfo(Rn).size;
+
+    size_t Rd_elem_size = vectorElementSize(detail->operands[0].vas);
+    int Rd_index = detail->operands[0].vector_index;
+    int Rn_index = detail->operands[1].vector_index;
+
+    uint32_t subRd = getVectorSubregister(Rd, Rn_size, Rd_index);
+    if (0 == subRd) {
+      LogError("%#lx: Invalid vector element access in operand 0",
+               instr->address);
+      return false;
+    }
+
+    if (-1 == Rn_index) {
+      /* INS (general) */
+      if (Rd_elem_size != Rn_size) {
+        LogError("%#lx: Operand size mismatch: Vd.Ts size %zu, Rn size %zu",
+                 instr->address, Rd_elem_size, Rn_size);
+        return false;
+      }
+
+      il.AddInstruction(
+          il.SetRegister(Rn_size, subRd, il.Register(Rn_size, Rn)));
+    } else {
+      /* INS (element) */
+
+      if (ARM64_VAS_INVALID == detail->operands[1].vas) {
+        LogError("%#lx: Vector operand 1 missing arrangement specifier",
+                 instr->address);
+        return false;
+      }
+
+      size_t Rn_elem_size = vectorElementSize(detail->operands[1].vas);
+      if (Rd_elem_size != Rn_elem_size) {
+        LogError("%#lx: Vector operand element size mismatch: %zu != %zu",
+                 instr->address, Rd_elem_size, Rn_elem_size);
+        return false;
+      }
+
+      uint32_t subRn = getVectorSubregister(Rn, Rn_size, Rd_index);
+      if (0 == subRn) {
+        LogError("%#lx: Invalid vector element access in operand 1",
+                 instr->address);
+        return false;
+      }
+
+      il.AddInstruction(il.SetRegister(Rd_elem_size, subRd,
+                                       il.Register(Rn_elem_size, subRn)));
+    }
+
+    return true;
+  }
+
+  bool LiftSTR(cs_insn* instr, LowLevelILFunction& il) {
+    cs_arm64* detail = &(instr->detail->arm64);
+
+    if (detail->op_count != 2 && detail->op_count != 3) {
+      return false;
+    }
+
+    if (detail->operands[0].type != ARM64_OP_REG ||
+        detail->operands[1].type != ARM64_OP_MEM) {
+      return false;
+    }
+
+    uint32_t Rt = this->m_base->GetRegisterByName(
+        cs_reg_name(disassembler.Get(), detail->operands[0].reg));
+
+    size_t Rt_size = this->m_base->GetRegisterInfo(Rt).size;
+
+    ExprId address = liftMemOperand(il, detail->operands[1]);
+    ExprId value;
+
+    if (ARM64_REG_WZR == detail->operands[0].reg ||
+        ARM64_REG_XZR == detail->operands[0].reg) {
```

Comment:
Unrelated to this PR, but this was a change I wanted to make in https://github.com/meme/aarch64-extensions/issues/1. For the future we should add a `ZeroOrRegister` function which returns the register OR a zero constant if the register is `wzr` or `xzr`.

Reply:
(up to you if you want to implement this change in this PR, but if so we'll have to follow up by placing that in the appropriate place in the other lifters)

Reply:
Let's do that all at once in another PR.

Reply:
Yep, sounds good.